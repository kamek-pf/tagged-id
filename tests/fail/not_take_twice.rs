use smartstring::alias::CompactString;
use tagged_id::{Id, Identify};

fn main() {
    struct TestStruct();

    impl Identify for TestStruct {
        type InnerId = CompactString;
    }

    let id1: Id<TestStruct> = "oops".into();
    let id2 = id1.take();

    assert_eq!(id1.take(), id2);
}
