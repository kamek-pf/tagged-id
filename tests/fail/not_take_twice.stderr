error[E0382]: use of moved value: `id1`
  --> tests/fail/not_take_twice.rs:14:16
   |
11 |     let id1: Id<TestStruct> = "oops".into();
   |         --- move occurs because `id1` has type `Id<TestStruct>`, which does not implement the `Copy` trait
12 |     let id2 = id1.take();
   |                   ------ `id1` moved due to this method call
13 |
14 |     assert_eq!(id1.take(), id2);
   |                ^^^ value used here after move
   |
note: `Id::<T>::take` takes ownership of the receiver `self`, which moves `id1`
  --> src/lib.rs
   |
   |     pub fn take(self) -> T::InnerId {
   |                 ^^^^
help: you can `clone` the value and consume it, but this might not be your desired behavior
   |
12 |     let id2 = id1.clone().take();
   |                  ++++++++
